### PR TITLE
feat(progressindicator): extend component features

### LINKED ISSUE #2605: [Progress Indicator] Upgrade PAL version for feature parity with Carbon
### What package is this for?

- [X] React
- [ ] Angular

### Summary

The [progress indicator from PAL](https://carbon-addons-iot-react.com/?path=/story/watson-iot-progressindicator--stateful) was created before the Carbon one. We should implement the Carbon version and deprecate the PAL version, as the Carbon version has "caught up"
[Carbon version](https://react.carbondesignsystem.com/?path=/story/components-progressindicator--default&knob-Collapsed%20icon%20text%20(tileCollapsedIconText)=Expand&knob-Collapsed%20icon%20text%20(tileExpandedIconText)=Collapse&knob-Form%20item%20(name%20in%20%3CTileGroup%3E)=tile-group&knob-Form%20item%20name%20(name%20in%20%3CRadioTile%3E)=tiles&knob-Form%20validation%20UI%20content%20(invalidText%20in%20%3CTimePicker%3E)=A%20valid%20value%20is%20required&knob-Href%20for%20clicable%20UI%20(href)=#&knob-Label%20text%20(labelText%20in%20%3CTimePicker%3E)=Select%20a%20time&knob-Label%20text%20(labelText%20in%20%3CTimePickerSelect%3E)=Please%20select&knob-Label%20text%20(labelText)=Checkbox%20label&knob-Max%20height%20(tileMaxHeight)=0&knob-Maximum%20length%20(maxLength%20in%20%3CTimePicker%3E)=5&knob-No%20label%20(hideLabel%20in%20%3CTimePickerSelect%3E)=true&knob-Plceholder%20text%20(placeholder%20in%20%3CTimePicker%3E)=hh:mm&knob-Regular%20expression%20for%20the%20value%20(pattern%20in%20%3CTimePicker%3E)=(1[012]|[1-9]):[0-5][0-9](\s)?(?i)&knob-Tab%20index%20(tabIndex)=0&knob-The%20title%20(title)=Section%201%20title&knob-Trigger%20icon%20description%20(iconDescription%20in%20%3CTimePickerSelect%3E)=open%20list%20of%20options)
Screen Shot 2021-07-15 at 9 45 23 AM

• All props from Carbon are exposed in PAL storybook
• Labels can display ellipsis and tooltip overflow
• Ensure keyboard nav (tab can navigate enter can trigger onChange)
• Migrate into [Wizard component](https://carbon-addons-iot-react.com/?path=/story/watson-iot-pagewizard--stateful-example) and ensure that component functions as it does today
• Create deprecation notice of PAL Progress Indicator (if necessary)

Depended on by https://jsw.ibm.com/browse/GRAPHITE-3838

### PR DESCRIPTION
**Pull request by kalyanixraut**
Closes #2605

**Summary**

- The optional `secondaryLabel` of ProgressIndicator step is displayed with ellipsis and native html tooltip on-hover when long text is provided, similar to step's label text.
- New boolean property `spaceEqually` is added, when set to `true` it makes ProgressIndicator steps to extend/adjust and spaced equally within available flex container.
- `spaceEqually` is exposed as a boolean propTypes in `PageWizard` component which uses ProgressIndicator to allow steps to be set space equally.

**Change List (commits, features, bugs, etc)**

- Altered `width` of optional label text to `width: auto` to enable ellipsis and native tooltip on-hover for lengthy text.
- Added optional `spaceEqually` prop to `ProgressIndicator` component. When set to `true` it will overwrite the `stepWidth` prop and add `flex-grow: 1` to all `li` elements(or steps) hence extending and spacing equally them within available flex container. This prop is needed in cases where there are fewer steps to display(like 2 or 3) and should be responsive with screen size.
- Added 'Steps spaced equally' story under `ProgressIndicator`.
- Added 'With steps spaced equally' story under `PageWizard`.
- Added unit test for `spaceEqually` prop `true` case in `ProgressIndicator` component.

**Acceptance Test (how to verify the PR)**

- Go to /?path=/story/1-watson-iot-progressindicator--presentation and ensure that long secondary step labels are displayed with ellipsis and tooltip with full text is displayed on-hover.
- Check the spaceEqually option under knobs section. The progress indicator steps should extend/adjust to cover the flex container and should be equally spaced. Resize the screen to check the progress indicator is responsive.
- Ensure the stepWidth option is not displayed when spaceEqually option is checked.
- Go to /?path=/story/1-watson-iot-pagewizard--with-sticky-footer-stateful-example-w-validation-in-page-title-bar and spaceEqually option is checked. Steps should adjust to the container width and should be equally spaced. Resize the screen to check it is responsive.
- Use the RTL knob to ensure everything adjusts correctly.

**Regression Test (how to make sure this PR doesn't break old functionality)**

- All remaining stories under ProgressIndicator and PageWizard works as before. Any other component story using ProgressIndicator should work as before.
- Go to /?path=/story/1-watson-iot-progressindicator--presentation ensure stepWidth prop should work as before when spaceEqually is not checked. 

https://deploy-preview-2622--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-progressindicator--presentation
https://deploy-preview-2622--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-pagewizard--with-sticky-footer-stateful-example-w-validation-in-page-title-bar

**Things to look for during review**

- [ ] Make sure all references to `iot` or `bx` class prefix is using the prefix variable
- [ ] (React) All major areas have a `data-testid` attribute. New test ids should have test written to ensure they are not changed or removed.
- [ ] UI should be checked in RTL mode to ensure the proper handling of layout and text.
- [ ] All strings should be translatable.
- [ ] The code should pass a11y scans (The storybook a11y knob should show no violations). New components should have a11y test written.
- [ ] Unit test should be written and should have a coverage of 90% or higher in all areas.
- [ ] All components should be passing visual regression test. For new styles or components either a visual regression test should be written for all permutations or the base image updated.
- [ ] Changes or new components should either write new or update existing documentation.
- [ ] PR should link and close out an existing issue

### CODE REVIEW

File: packages/react/src/components/PageWizard/PageWizard.story.jsx [hunk 1]
```diff
@@ -410,6 +410,27 @@ export const WithHorizontalProgressIndicator = () => (
 
 WithHorizontalProgressIndicator.storyName = 'With Horizontal ProgressIndicator';
 
+export const WithSpaceEqually = () => (
```
Comment (davidicus) [inline packages/react/src/components/PageWizard/PageWizard.story.jsx:413]: Can we just make this a knob on an existing story? I don't think it needs a story of its own.

File: packages/react/src/components/ProgressIndicator/ProgressIndicator.story.jsx [hunk 2]
```diff
@@ -132,6 +132,45 @@ Skeleton.parameters = {
...
+    id: 'step3',
+    label: 'Third Step',
+    secondaryLabel: 'Optional label',
+    disabled: true,
+  },
+  { id: 'step4', label: 'Fourth Step', invalid: true },
+  { id: 'step5', label: 'Fifth Step', secondaryLabel: 'Optional label is very, very long' },
+];
+
+export const SpaceEqually = () => {
```
Comment (davidicus) [inline packages/react/src/components/ProgressIndicator/ProgressIndicator.story.jsx:157]: Same comment as above. Can we just include this prop in an existing story with a knob that can control it?

**Review by davidicus — COMMENTED** [review]
Looks great. There is just the comment about the stories and this should be good to go.

File: packages/react/src/components/PageWizard/PageWizard.story.jsx (hunk 1 above)
Comment (kalyanixraut) [reply packages/react/src/components/PageWizard/PageWizard.story.jsx]: Done. Removed the story and added to knob https://deploy-preview-2622--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-pagewizard--with-sticky-footer-stateful-example-w-validation-in-page-title-bar

File: packages/react/src/components/ProgressIndicator/ProgressIndicator.story.jsx (hunk 2 above)
Comment (kalyanixraut) [reply packages/react/src/components/ProgressIndicator/ProgressIndicator.story.jsx]: Done. Removed the story and added knob to https://deploy-preview-2622--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-progressindicator--presentation

**Review by davidicus — APPROVED** [review]

**Review by stuckless — APPROVED** [review]

**Pull request merged**